tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
+        "[General]\n",
+        "  riverkm     = RiverKM\n",
+        "  figuredir   = FigureDir\n",
+        "  outputdir   = OutputDir\n",
+        "\n",
+        "[SomeSection]\n",
+        "  reference   = reference\n",
+        "  withmeasure = with_measure\n",
+        ]
+        
+        file_path = tmp_path / "test_file.cfg"
+        config = self.sample_config(tmp_path)
+        
+        dfastmi.batch.save_configuration_file(file_path, config)
+        
+        assert os.path.exists(file_path)
+        with open(file_path, 'r') as file:
+            file_lines = file.readlines()
+            
+        assert len(file_lines) == len(expected_lines)
+        assert file_lines == expected_lines
+        
+    def sample_config(self, tmp_path):
+        # Create a sample configuration
+        config = ConfigParser()
+        config["General"] = {
+            "RiverKM": tmp_path / "RiverKM",
+            "FigureDir": tmp_path / "FigureDir",
+            "OutputDir": tmp_path / "OutputDir"
+        }
+        config["SomeSection"] = {
+            "Reference": tmp_path / "reference",
+            "WithMeasure": tmp_path / "with_measure"
+        }
+        return config
+    
+class Test_batch_countq():       
+    @pytest.mark.parametrize("vector_data, expected_non_empty_discharges_count", [
+        ([0.123, None, 0.456, 0.789, None], 3),
+        ([0.123, 0.123, 0.456, 0.789, 0.123], 5),
+        ([None, None, None, None, None], 0),
+    ])    
+    def given_vector_with_discharges_when_countq_then_return_amount_of_non_empty_discharges(self, vector_data: Vector, expected_non_empty_discharges_count: int):
+        assert dfastmi.batch.countQ(vector_data) == expected_non_empty_discharges_count
+ 
+class Test_batch_write_report():   
+    @pytest.mark.parametrize("slength", [
+        0.202122,
+        1.202122
+    ])       
+    def given_input_data_with_varying_slength_when_write_report_then_expect_messages_written_in_report(self, slength :float):
+        report = StringIO()
+        reach = "reach"
+        q_location = "location"
+        q_threshold = 0.123
+        q_bankfull = 0.456
+        q_stagnant = 0.789
+        tstag = 0.101112
+        q_fit = [0.131415, 0.171819]
+        Q = [0.232425, 0.262728, 0.293031]
+        T = [0.323334, 0.353637, 0.383940]
+        
+        ApplicationSettingsHelper.PROGTEXTS.clear()
+        
+        dfastmi.batch.write_report(report, reach, q_location, q_threshold, q_bankfull, q_stagnant, tstag, q_fit, Q, T, slength)
+
+        report_lines = report.getvalue().split('\n')
+                
+        prefix = 'No message found for '
+        assert prefix + 'reach' in report_lines
+        assert prefix + 'report_qthreshold' in report_lines
+        assert prefix + 'report_qbankfull' in report_lines
+        assert prefix + 'closed_barriers' in report_lines
+        assert prefix + 'char_discharge' in report_lines
+        assert report_lines.count(prefix + 'char_discharge') is 3
```

Comment:
why did you use the word **is** i thought for datatypes (int, etc) we use == (or != for invert) and for types **is** like:

assert Marlon is Human
assert Marlon is AltenEmployee
assert Marlon is not Stupid

assert Marlon.Age == 27
assert Marlon.Salary == 4000
assert Marlon.HourlyRate == 87.50

assert Marlon.IsFunToWorkWith

Reply:
I do not know why I used is, I will update it.

unfortunately some of the above described asserts are failing :sob:

Reply:
done